lib/reverse_sentence.rb

```diff
@@ -1,6 +1,61 @@
+# helper method
+# Time complexity: O(n)
+# Space complexity: O(n)
+def special_split(string)
```

Comment:
This seems an unnecessarily complicated way to split up the string into an array of words/spaces.

Your solution works, but it could be done with O(1) space complexity.  Consider reversing the string letter-by-letter, and then going through and reversing each word using the indexes.